pkgs/development/python-modules/filedate/default.nix

```
@@ -0,0 +1,58 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pip,
+  python-dateutil,
+}:
+buildPythonPackage rec {
+  pname = "filedate";
+  version = "3.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "kubinka0505";
+    repo = pname;
+    rev = version;
+    hash = "sha256-HvuGP+QlUlfAUfFmaVVvtPHGdrbWVxghQipnqTTvAQc=";
+  };
+
+  unpackPhase = ''
+    cp -r $src/Files/* .
+    cp -r $src/ReadMe.md .
+  '';
+
+  # The repo stores everything in "src" and uses setup.py to move "src" ->
+  # "filedate" before calling setup() and then tries to rename "filedate" back
+  # to "src" after.
+  postPatch = ''
+    mv src filedate
+    substituteInPlace setup.py \
+      --replace-fail "__title__ = os.path.basename(os.path.dirname(os.path.dirname(__file__)))" '__title__ = "filedate"'
+    substituteInPlace setup.py \
+      --replace-fail "cleanup = True" "cleanup = False"
+
+    # Read the ReadMe.md in our patched location
+    substituteInPlace setup.py \
+      --replace-fail 'desc = open("../ReadMe.md", encoding = "UTF-8").read().rstrip()' 'desc = open("ReadMe.md", encoding = "UTF-8").read().rstrip()'
+
+    # Disable renaming "filedate" dir back to "src"
+    substituteInPlace setup.py \
+      --replace-fail "if os.path.exists(__title__):" ""
+    substituteInPlace setup.py \
+      --replace-fail "	os.rename(__title__, directory)" ""
+  '';
+
+  buildInputs = [ pip ];
+
+  propagatedBuildInputs = [ python-dateutil ];
+
+  pythonImportsCheck = [ "filedate" ];
+
+  meta = {
+    description = "Simple, convenient and cross-platform file date changing library.";
+    homepage = "https://github.com/kubinka0505/filedate";
```

```suggestion
    homepage = "https://github.com/kubinka0505/filedate";
    changelog = "https://github.com/kubinka0505/filedate/blob/${src.rev}/Documents/ChangeLog.md";
```
Adding meta.changelog is useful for future updates.

---

Nice catch. Updated.